test/WireMock.Net.Tests/Matchers/JsonPartialMatcherTests.cs

```diff
@@ -0,0 +1,360 @@
+                { "Name", new JValue("Test") }
+            };
+            double match = matcher.IsMatch(jobject);
+
+            // Assert 
+            Assert.Equal(1.0, match);
+        }
+
+        [Fact]
+        public void JsonPartialMatcher_IsMatch_JObjectAsString_RejectOnMatch()
+        {
+            // Assign 
+            var matcher = new JsonPartialMatcher(MatchBehaviour.RejectOnMatch, "{ \"Id\" : 1, \"Name\" : \"Test\" }");
+
+            // Act 
+            var jobject = new JObject
+            {
+                { "Id", new JValue(1) },
+                { "Name", new JValue("Test") }
+            };
+            double match = matcher.IsMatch(jobject);
+
+            // Assert 
+            Assert.Equal(0.0, match);
+        }
+
+        [Fact]
+        public void JsonPartialMatcher_IsMatch_JObjectWithDateTimeOffsetAsString()
+        {
+            // Assign 
+            var matcher = new JsonPartialMatcher("{ \"preferredAt\" : \"2019-11-21T10:32:53.2210009+00:00\" }");
+
+            // Act 
+            var jobject = new JObject
+            {
+                { "preferredAt", new JValue("2019-11-21T10:32:53.2210009+00:00") }
+            };
+            double match = matcher.IsMatch(jobject);
+
+            // Assert 
+            Assert.Equal(1.0, match);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidMatches))]
+        public void JsonPartialMatcher_IsMatch_StringInputValidMatch(string value, string input)
+        {
+            // Assign
+            var matcher = new JsonPartialMatcher(value);
+
+            // Act
+            double match = matcher.IsMatch(input);
+
+            // Assert
+            Assert.Equal(1.0, match);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidMatches))]
+        public void JsonPartialMatcher_IsMatch_StringInputWithInvalidMatch(string value, string input)
+        {
+            // Assign
+            var matcher = new JsonPartialMatcher(value);
+
+            // Act
+            double match = matcher.IsMatch(input);
+
+            // Assert
+            Assert.Equal(0.0, match);
+        }
+
+        public static IEnumerable<object[]> ValidMatches()
+        {
+            yield return new object[] { "{\"test\":\"abc\"}", "{\"test\":\"abc\",\"other\":\"xyz\"}" };
+            yield return new object[] { "\"test\"", "\"test\"" };
+            yield return new object[] { "123", "123" };
+            yield return new object[] { "[\"test\"]", "[\"test\"]" };
+            yield return new object[] { "[\"test\"]", "[\"test\", \"other\"]" };
+            yield return new object[] { "[123]", "[123]" };
+            yield return new object[] { "{ \"test\":\"value\" }", "{\"test\":\"value\",\"other\":123}" };
+            yield return new object[] { "{ \"test\":\"value\" }", "{\"test\":\"value\"}" };
+            yield return new object[] { "{ \"test.nested\":\"value\" }", "{\"test\":{\"nested\":\"value\"}}" };
```

Comment:
How is it possible that a dot-notation equals the nested value?

Reply:
Again, the idea here is to allow shorthand matching against nested objects.
E.g.: **{ "test.nested": "abc" }** matches against { "test": { "nested": "abc" } }

The property here, essentially, is treated as a JPath, like in the JSONPathParser.

I've renamed the JsonMatcher/JsonPartialMatcher.AreEqual protected method into IsMatch to avoid the confusion.

Reply:
Can you please create a separate Unit test which only tests these shorthand notations. (because I guess I will forget this logic in a year from now why this works...)

Reply:
Good point. Moved JPath tests into distinct tests. Hope it's better now.